jobs/async-upload/tests/test_config.py

```diff
@@ -258,3 +273,135 @@ def test_params_will_override_env_config(
+            "--source-uri",
+            uri_value,
+            "--registry-server-address",
+            "https://registry.example.com",
+        ]
+    )
+
+    assert isinstance(config.source, URISourceStorageConfig)
+    assert config.source.uri == uri_value
+    assert config.source.credentials_path is None
+
+    assert isinstance(config.destination, OCIStorageConfig)
+    assert config.destination.uri == destination_oci_env_vars["oci_uri"]
+
+
+def test_uri_credentials_override_params(uri_credentials_folder, destination_oci_env_vars, model_env_vars):
+    """Test that URI credentials from file override CLI parameters"""
+    folder_location, expected_credentials = uri_credentials_folder
+    cli_uri = "hf://cli/model/override"
+
+    config = get_config(
+        [
+            "--source-type",
+            "uri",
+            "--source-uri",
+            cli_uri,
+            "--source-uri-credentials-path",
+            str(folder_location),
+            "--registry-server-address",
+            "https://registry.example.com",
+        ]
+    )
+
+    # Credentials from file should override CLI parameter
+    assert isinstance(config.source, URISourceStorageConfig)
+    assert config.source.uri == expected_credentials["uri"]  # From file, not CLI
+    assert config.source.credentials_path == str(folder_location)
+
+
+def test_uri_credentials_missing_folder_error(model_env_vars, destination_oci_env_vars):
+    """Test that missing credentials folder raises appropriate error"""
+    with pytest.raises(FileNotFoundError, match="credentials folder not found"):
+        get_config(
+            [
+                "--source-type",
+                "uri",
+                "--source-uri-credentials-path",
+                "/non/existent/path",
+                "--registry-server-address",
+                "https://registry.example.com",
+            ]
+        )
+
+
+def test_uri_credentials_missing_uri_file_error(model_env_vars, destination_oci_env_vars, tmp_path):
+    """Test that missing URI file in credentials folder raises appropriate error"""
+    # Create empty directory without URI file
+    with pytest.raises(FileNotFoundError, match="URI credential file not found"):
+        get_config(
+            [
+                "--source-type",
+                "uri",
+                "--source-uri-credentials-path",
+                str(tmp_path),
+                "--registry-server-address",
+                "https://registry.example.com",
+            ]
+        )
+
+
+def test_uri_credentials_env_var_support(model_env_vars, destination_oci_env_vars, tmp_path, monkeypatch):
+    """Test that URI credentials path can be set via environment variable"""
+    # Create URI credential file
+    uri_value = "hf://env/test/model"
+    uri_file = tmp_path / "URI"
+    uri_file.write_text(uri_value)
+
+    with monkeypatch.context() as mp:
+        # Set environment variable
+        mp.setenv("MODEL_SYNC_SOURCE_URI_CREDENTIALS_PATH", str(tmp_path))
```

Comment:
shouldn't this use

```py
folder_location, expected_credentials = uri_credentials_folder
```

with `uri_credentials_folder` fixture and then setenv value to `folder_location` ? 🤔

Reply:
+1